types/trigger.d.ts

```diff
@@ -143,7 +143,12 @@ export type DataInitializeFunc<Data extends RaidbossData> = () => Omit<Data, key
 
 export type DisabledTrigger = { id: string; disabled: true };
 
-export type TriggerSet<Data extends RaidbossData> = {
+type Required<Type> =
+  {
+    [key in keyof Type]-?: Record<string, never> extends Pick<Type, key> ? never : key
```

Comment:
Can this get a comment as to what it's doing? I don't think I've ever seen `-?` syntax before.

Reply:
Comment added, I think it explains things pretty thoroughly?